### ISSUE TITLE
`CFLAGS`: `SC2034` false positive, optionally whitelist env vars

### ISSUE THREAD

**Issue by Neved4** [koalaman/shellcheck#2878]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version: `0.9.0`
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2034)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

CFLAGS=-O2
cc -o foo foo.c
cc -o bar bar.c
```

#### Here's what `shellcheck` currently says:
```log
Line 3:
CFLAGS=-O2
^-- SC2034 (warning): CFLAGS appears unused. Verify use (or export if used externally).
```

#### Here's what I wanted or expected to see:

No warnings, `gcc`/`clang` implicitly use this variable.

Would it make sense that `shellcheck` whitelists `CFLAGS` and related variables?
There are other similar, common ones, present in [POSIX 2017.1: _Vol 1. Base Definitions_, _§8. Environment Variables_](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html)

**Comment by josephcsible** [comment]
Are you sure that's a false positive? I'm pretty sure it's a true positive and `cc` doesn't actually see that variable.

**Comment by Neved4** [comment]
@josephcsible You're right, it's a true positive within the context of the script! 🤍

Maybe there are prod/dev environments where `export CFLAGS` and other vars are commonly set. There, the vars are not really being unused, but getting used implicitly.

Perhaps `shellcheck` could be aware of the existing environment? Feel free to close.
